src/main/java/seedu/allonus/storage/StorageFile.java

```diff
@@ -1,4 +1,214 @@
 package seedu.allonus.storage;
 
+import seedu.allonus.contacts.ContactsManager;
+import seedu.allonus.contacts.entry.Contact;
+import seedu.allonus.expense.Expense;
+import seedu.allonus.expense.ExpenseTracker;
+import seedu.allonus.modules.Module;
+import seedu.allonus.modules.StudyManager;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+
 public class StorageFile {
-}
+    private static ContactsManager contactsManager;
+    private static StudyManager studyManager;
+    private static ExpenseTracker expenseTracker;
+
+    public static final String DIRECTORY_NAME = "data";
+    public static final String DATAFILE_RELATIVE_PATH = DIRECTORY_NAME + File.separator + "allonusData.txt";
+    public static final String PROJECT_PATH = System.getProperty("user.dir");
+
+    public StorageFile() {
+    }
+
+    public static void setFields(ContactsManager contactsManagerParam, ExpenseTracker expenseTrackerParam,
+                                 StudyManager studyManagerParam) {
+        contactsManager = contactsManagerParam;
+        expenseTracker = expenseTrackerParam;
+        studyManager = studyManagerParam;
+    }
+
+    public void createDirectory() {
+        String directory = DIRECTORY_NAME;
+        String directoryPath = PROJECT_PATH + File.separator + directory;
+        File dir = new File(directoryPath);
+        boolean isMade = dir.mkdir();
+        if (!isMade) {
+            System.out.println("Error creating the new directory.");
+        }
+    }
+
+    /**
+     * This method creates a file at path specified in <code>Duke.DATAFILE_RELATIVE_PATH</code>. A directory is
+     * created first if it doesn't exist.
+     *
+     * @see IOException
+     * @see File
+     */
+    public void createFile() {
+        File dir = new File(DIRECTORY_NAME);
+        // attempt to create directory if it does not exist.
+        while (!dir.exists()) {
+            createDirectory();
+        }
+        File f = new File(PROJECT_PATH + File.separator + DATAFILE_RELATIVE_PATH);
+
+        try {
+            f.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Error creating the new file.");
+        }
+    }
+
+    public void loadExpense(String expense) {
+        expenseTracker.loadAdd(expense);
+    }
+
+    public void loadModule(String module) {
+        studyManager.addModule(module);
```

Comment:
I like the way you have reused the existing functions defined under the respective managers to add items. Good application of abstraction!